### PR TITLE
Drafting docs for noir js

### PR DESCRIPTION
**Pull request by signorecello**

### CODE REVIEW

File: docs/noir_js/noir_js.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    Noir project,
+    javascript,
+    typescript,
+    node.js,
+    browser,
+    react
+  ]
+---
+
+Noir JS are a set of typescript libraries that make it easy to use Noir on your dapp, webapp, node.js server, website, etc.
```
Comment (critesjosh) [inline docs/noir_js/noir_js.md:16]: Is this one package or multiple? If its one, it should say "Noirjs is a set..."

**Comment by critesjosh** [comment]
Do you auto-generate the files in `reference/`? If so, how does that work? It would probably be good to add to the docs README, so others can update these if needed.

**Comment by signorecello** [comment]
I didn't auto generate... Yeah I'm crazy, but I'm a crazy guy in a hurry to get this shipped 🚀 We should totally use JSDocs or something to autogenerate this

File: docs/noir_js/noir_js.md (hunk 1 above)
Comment (signorecello) [reply docs/noir_js/noir_js.md:16]: Multiple packages for now (@noir-lang/backend_barretenberg and @noir-lang/noir_js). Probably they'll always be two packages, but we're still discussing this

File: docs/noir_js/getting_started/01_setup.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+description:
+  Learn how to setup a new app that uses Noir to generate and verify zero-knowledge SNARK proofs in a typescript or javascript environment
+keywords: [how to, guide, javascript, typescript, noir, barretenberg, zero-knowledge, proofs]
+---
+
+Noir JS works both on the browser and on the server, and works for both ESM and CJS module systems. In this page, we will learn how can we write a simple test and a simple webapp to verify the Standard Noir Example.
+
+## Before we start
+
+Make sure you have Node installed on your machine by opening a terminal and executing `node --version`. If you don't see a version, you should install [node](https://nodejs.org/en/download). You can also use `yarn` if you prefer that package manager.
```
Comment (critesjosh) [inline docs/noir_js/getting_started/01_setup.md:12]: Is this your preferred way to install node? I usually use nvm, https://github.com/nvm-sh/nvm

Comment (critesjosh) [reply docs/noir_js/getting_started/01_setup.md:12]: we could also suggest bun :grin:

File: docs/noir_js/reference/01_bb_backend.md [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+  Number, (optional)
+  The number of threads to be used by the backend. Defaults to 1.
+
+### Usage
+
+```js
+const backend = new BarretenbergBackend(acirCircuit);
+```
+
+## instantiate
```
Comment (critesjosh) [inline docs/noir_js/reference/01_bb_backend.md:56]: Should this method be included since it can't be called by devs?

**Pull request merged**